lms/resources/_js_config/__init__.py

```diff
@@ -104,8 +104,8 @@ def asdict(self):
 
     def enable_oauth2_redirect_error_mode(
         self,
-        error_details,
-        auth_url=None,
+        auth_route,
+        error_details="",
```

Comment:
I change this to take an auth _route_ and generate the URL itself. This allows the `route_url()` call and also the `BearerTokenSchema` call to live in this method and be re-used for both Blackboard and Canvas.

Also made `error_details` optional because Blackboard doesn't use it

Reply:
I was wondering if the `error_details`, `is_scope_invalid` and `canvas_scopes` are all essentially Canvas specific. Seeing as the caller picks which template this is being rendered in, but looking at the template none of them are used in it, so it seems reasonable to control the values.

Otherwise I was going to suggest this could be `**template_kwargs` or something and they just get passed through.